### PR TITLE
Upgrade / align with dekorate 2.0.0

### PR DESCRIPTION
**Pull request by iocanel**
Dekorate 2.0.0 brings a lot of internal changes, this pull request aligns things.

Most noteworthy changes that are related to Quarkus:

# Coding & style
- No more references to shaded classes.
- Naming is more specific (e.g. Handler -> ManifestGenerator).
- Separate concerns (better isolate apt from implementation & split service from factories).

# Behavior
- Mapping of ports is not done in the Service instead of Ingress Route (#13945)
- VCS annotations no longer sanitize urls (#14827)

# Potential breaking changes
#13945 does bring in a change that may affect users that are referencing the generated Service using `http://<service name>:<container port>`. Since, the container port of http services is now mapped to port 80, the port part will either have to be removed, or replaced with value '80' instead of the actual container port.

Users that want the old behavior (Service without mapping), will have to explicitly configure the port of the Service:

    quarkus.kubernetes.ports.http.name=http
    quarkus.kubernetes.ports.http.host-port=8080
    quarkus.kubernetes.ports.http.container-port=8080

### CODE REVIEW

**Comment by geoand** [comment]
Seems to need a rebase right off the bat :)

**Comment by geoand** [comment]
@famod is the `skipped` in some checks expected?

**Comment by gsmet** [comment]
It's because I cancelled the incremental builds because of the load.

**Comment by geoand** [comment]
Understood, thanks

**Comment by geoand** [comment]
@iocanel can you add in the description of the PR what the general themes of dekorate 2.0 that relate to these changes Quarkus are?

Thanks

**Comment by iocanel** [comment]
> @iocanel can you add in the description of the PR what the general themes of dekorate 2.0 that relate to these changes Quarkus are?
> 
> Thanks

Done

**Comment by geoand** [comment]
I assume the linked issue would be closed by this PR right?

If so, then please use the `Fixes` or `Resolves` syntaxt so they will be closed automatically when we merge this.

**Comment by geoand** [comment]
>     * Mapping of ports is not done in the Service instead of Ingress Route (#13945)

Can you paste an example of the behavior before and after this PR WRT the above? 
I'd to be able to see it in action

**Comment by iocanel** [comment]
> I assume the linked issue would be closed by this PR right?
> 
> If so, then please use the `Fixes` or `Resolves` syntaxt so they will be closed automatically when we merge this.

No, I intend to create followup pull request that will add tests and then resolve the issues.

**Comment by geoand** [comment]
Understood, thanks

File: integration-tests/kubernetes/maven-invoker-way/src/it/openshift-s2i-build-and-deploy/verify.groovy [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import io.dekorate.utils.Serialization
-import io.dekorate.deps.kubernetes.api.model.KubernetesList
-import io.dekorate.deps.kubernetes.api.model.apps.Deployment;
-import io.dekorate.deps.openshift.api.model.*
+import io.fabric8.kubernetes.api.model.KubernetesList
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.openshift.api.model.*
```
Comment (geoand) [inline integration-tests/kubernetes/maven-invoker-way/src/it/openshift-s2i-build-and-deploy/verify.groovy:4]: Please avoid all star imports

**Comment by geoand** [comment]
We just need a rebase for this. I only had one tiny comment

**Comment by iocanel** [comment]
> > ```
> > * Mapping of ports is not done in the Service instead of Ingress Route (#13945)
> > ```
> 
> Can you paste an example of the behavior before and after this PR WRT the above?
> I'd to be able to see it in action

# Before 

Iniitially services used to be pass through to the actual container port e.g. 8080 -> 8080
 
```
kind: Service
metadata:
  name: my-service
spec:
  ports:
  - name: http
    port: 8080
    targetPort: 8080
  selector:
    app.kubernetes.io/name: my-example
    app.kubernetes.io/version: 1.0
  type: ClusterIP
```

# After

Http ports are mapped to port 80. This is something that was done on Ingreess/Route level, but now its done on Service.

```
kind: Service
metadata:
  name: my-service
spec:
  ports:
  - name: http
    port: 80
    targetPort: 8080
  selector:
    app.kubernetes.io/name: my-example
    app.kubernetes.io/version: 1.0
  type: ClusterIP
```

**Comment by iocanel** [comment]
Again, I wouldn't sweat it on the underlying issues, since there is no intention to close them in this PR.
Let's keep the PR focused on the actual code changes required and we will deal with the other issues in followup PRs.

**Comment by geoand** [comment]
The Service change makes sense, but it is a breaking change is it not?

I mean it changes stuff for people who deploy the application but don't expose it as they intend to make it available only internally.

**Comment by iocanel** [comment]
> The Service change makes sense, but it is a breaking change is it not?
> 
> I mean it changes stuff for people who deploy the application but don't expose it as they intend to make it available only internally.

Indeed this is a breaking change. 
One thing we could do is to try and maintain backwards compatibility by using two port mappings where it make sense (if the it's supported by kubernetes).

**Comment by geoand** [comment]
> > The Service change makes sense, but it is a breaking change is it not?
> > I mean it changes stuff for people who deploy the application but don't expose it as they intend to make it available only internally.
> 
> Indeed this is a breaking change.
> One thing we could do is to try and maintain backwards compatibility by using two port mappings where it make sense (if the it's supported by kubernetes).

Or have some kind of setting that allows reverting to the old behavior?

**Comment by geoand** [comment]
> > > > The Service change makes sense, but it is a breaking change is it not?
> > > > I mean it changes stuff for people who deploy the application but don't expose it as they intend to make it available only internally.
> > > 
> > > 
> > > Indeed this is a breaking change.
> > > One thing we could do is to try and maintain backwards compatibility by using two port mappings where it make sense (if the it's supported by kubernetes).
> > 
> > 
> > Or have some kind of setting that allows reverting to the old behavior?
> 
> Users are already able to set the actual port using the existing configuration options. The mapping really applies when no port has been explicitly configured.
> 

But this is likely the most common case, no? 
If we don't do anything more than this PR, we at least need to put into the migration guide the **exact** quarkus configuration users need to set in order to revert to the old behavior.

**Comment by iocanel** [comment]
Since, we've gone deep into the discussion about Service port mapping. I think that it would be better if I included tests, docs etc directly in this PR. wdyt @geoand ?

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by geoand** [comment]
@iocanel can you write something up for us to add to the migration guide of Quarkus 2.x?

This is a breaking change but we don't have an entry in https://github.com/quarkusio/quarkus/wiki/Migration-Guide-2.0

**Pull request merged**